AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,422 @@
+// swiftlint:disable file_length
+class DataStoreConsecutiveUpdatesTests: SyncEngineIntegrationTestBase {
+    /// - Given: API has been setup with `Post` model registered
+    /// - When: A Post is saved and then immediately updated
+    /// - Then: The post should be updated with new fields immediately and in the eventual consistent state
+    func testSaveAndImmediatelyUpdate() throws {
+        try startAmplifyAndWaitForSync()
+
+        // create a post
+        let myPost = Post(id: UUID().uuidString,
+                          title: "MyPost",
+                          content: "This is my post.",
+                          createdAt: Temporal.DateTime.now(),
+                          rating: 3,
+                          status: .published)
+
+        var updatedPost = myPost
+        updatedPost.rating = 5
+        updatedPost.title = "MyUpdatedPost"
+        updatedPost.content = "This is my updated post."
+
+        let saveSyncExpectation = expectation(description: "Post is saved and synced")
+        let updateSyncExpectation = expectation(description: "Post is updated and synced")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent
+            else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post,
+                  post.id == myPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post.title, myPost.title)
+                XCTAssertEqual(post.content, myPost.content)
+                XCTAssertEqual(post.rating, myPost.rating)
+                saveSyncExpectation.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
+                // this should be removed once the bug is fixed
+                XCTAssertEqual(post.title, myPost.title)
+                XCTAssertEqual(post.content, myPost.content)
+                XCTAssertEqual(post.rating, myPost.rating)
+
+                // this is the expected behavior which is currently failing
+                // XCTAssertEqual(post.title, "MyUpdatedPost")
+                // XCTAssertEqual(post.content, "This is my updated post.")
+                // XCTAssertEqual(post.rating, 5)
+
+                updateSyncExpectation.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post
+        let immediateSaveExpectation = expectation(description: "Post is saved")
```

Comment:
"immediately" and "eventual state" are scenarios, the name of the expectation with "immediate" doesn't really provide any context for the scenario here. if the scenario is trying to save and then update immediately then i'd suggest to nest the call and name the expectation "saveAndImmediatelyUpdate"
```swift
let saveAndImmediatelyUpdate = expectation(description: "Post is saved and then immediately updated")
Amplify.DataStore.save(newPost) { result in 
   switch result {
    case .success: 
          Amplify.DataStore.save(updatedPost) { result in
          switch result {
           case .success:
               saveAndImmediatelyUpdateExpectation.fulfill()
````